dimscord/objects/typedefs.nim

```diff
@@ -899,3 +900,10 @@ proc `$`*(e: Emoji): string =
             e.name.get("?") & ":" & e.id.get
         else:
             e.name.get("?")
+
+
+template api*[T: Context](ctx: T): RestApi =
+    result = 
+        when ctx is Shard: ctx.client.api
+        elif ctx is DiscordClient: ctx.api
+        elif ctx is RestApi: ctx
```

Comment:
```suggestion
    when ctx is Shard: ctx.client.api
    elif ctx is DiscordClient: ctx.api
    elif ctx is RestApi: ctx
```

Since templates just replace where they are called, the `result` variable will be the `result` where its called.
Having it this way means it will just return the `RestApi` object